airbyte-commons/src/main/java/io/airbyte/commons/json/JsonSchemas.java

```diff
@@ -189,8 +189,7 @@ private static void traverseJsonSchemaInternal(final JsonNode jsonSchemaNode,
             // hit every node.
             traverseJsonSchemaInternal(jsonSchemaNode.get(JSON_SCHEMA_ITEMS_KEY), newPath, consumer);
           } else {
-            throw new IllegalArgumentException(
-                "malformed JsonSchema array type, must have items field in " + jsonSchemaNode);
+            log.warn("The array is missing an items field. The traversal is silently stopped. Current schema: " + jsonSchemaNode);
```

Comment:
Nit:  You can just use the placeholders in SLF4:  `Current schema: {}", jsonSchemaNode);`